File: 1-Two-Sum.scala
import scala.collection.mutable

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val map = mutable.Map.empty[Int, Int]
        
        for (i <- nums.indices) {
            val complement = target - nums(i)
            if (map.contains(complement)) {
                return Array(map(complement), i)
            }
            map(nums(i)) = i
        }
        
        Array.empty[Int]
    }
}

File: neetcode/1-Two-Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const map: Record<number, number> = {}

  for (let i = 0; i < nums.length; i++) {
    const dif = target - nums[i]

    if (map[dif] !== undefined) return [map[dif], i]
    map[nums[i]] = i
  }

  return [-1]
};
